### PR TITLE
fix(api, class, style): remove deprecated selectors

### LINKED ISSUE #410: Error "Can't bind to 'ngStyle.md' since it isn't a known property of 'div'"
Error at runtime.

Should there be a [ngStyle.md] selector in style.ts? There is one for [style.md].

### PR DESCRIPTION
**Pull request by ThomasBurleson**
* all `[style]`, `[style.xxx]`,`[class]`, and `[class.xxx]` selectors have been removed
  * only `ngStyle` and `ngClass` selectors support responsive suffices
  * now use of raw `style` attribute will NOT instantiate a StyleDirective
  * now use of raw `class` attribute will NOT instantiate a ClassDirective
* The `ClassDirective` now simply adds responsive enhancements to the core `NgClass` directive
* The `StyleDirective` now simply adds responsive enhancements to the core `NgStyle` directive
* Added missing `[ngStyle.md]` selector
* Added `isBrowser()` universal checks for `getAttribute()` and `getStyle()` queries

Fixes #410, #408, #273, #420. Closes #418.

BREAKING CHANGES

* **ngStyle**, **ngClass** should be used for responsive selectors and changes. Raw `style` and `class` attributes no longer support responsive suffices.

##### Before

```html
<div  fxLayout
  [class.xs]="['xs-1', 'xs-2']"
  [style]="{'font-size': '10px', 'margin-left' : '13px'}"
  [style.xs]="{'font-size': '16px'}"
  [style.md]="{'font-size': '12px'}">
</div>
```

##### Now

```html
<div  fxLayout
  [ngClass.xs]="['xs-1', 'xs-2']"
  [ngStyle]="{'font-size': '10px', 'margin-left' : '13px'}"
  [ngStyle.xs]="{'font-size': '16px'}"
  [ngStyle.md]="{'font-size': '12px'}">
</div>
```

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
@mmalerba - ready for your review and g3 testing.

File: src/lib/api/core/base-adapter.ts [hunk 1]
```diff
@@ -20,6 +20,16 @@ import {MediaMonitor} from '../../media-query/media-monitor';
 export class BaseFxDirectiveAdapter extends BaseFxDirective {
 
   /**
+   * Does this directive have 1 or more responsive keys defined
+   * Note: we exclude the 'baseKey' key (which is NOT considered responsive)
+   */
+  public hasResponsiveAPI() {
+    const totalKeys = Object.keys(this._inputMap).length;
+    const baseValue = this._inputMap[this._baseKey];
+    return  (totalKeys - (!!baseValue ? 1 : 0)) > 0;
```
Comment (mmalerba) [inline src/lib/api/core/base-adapter.ts]: can this just be `return this.hasResponsiveAPI(this._baseKey)`?

Comment (ThomasBurleson) [reply src/lib/api/core/base-adapter.ts]: Sorry @mmalerba - I do not understand your question here ^.

Comment (ThomasBurleson) [reply src/lib/api/core/base-adapter.ts]: the `_baseKey` is the only non-responsive key. We want to check if 1 or more responsive keys are defined/in-use.

Comment (mmalerba) [reply src/lib/api/core/base-adapter.ts]: I'm just saying that this class extends a class with a 1-parameter method of the same name, it looks like this no-parameter version could just call the 1-parameter version

Comment (ThomasBurleson) [reply src/lib/api/core/base-adapter.ts]: Fixed.

File: src/lib/api/ext/class.ts [hunk 2]
```diff
@@ -154,51 +124,50 @@ export class ClassDirective extends BaseFxDirective
...
-   *  Identify the activated input value and update the ngClass iterables...
-   *  needs ngDoCheck() to actually apply the values to the element
+   * Initial lookup of raw 'class' value (if any)
    */
-  protected _updateNgClass(value?: NgClassType) {
-    if (this._ngClassAdapter.mqActivation) {
-      value = this._ngClassAdapter.mqActivation.activatedInput;
+  protected _fallbackToKlass() {
+    if (!this._base.queryInput('ngClass')) {
+      this.ngClassBase = this._getAttributeValue('class');
```
Comment (ThomasBurleson) [inline src/lib/api/ext/class.ts]: Premise: When a breakpoint deactivates, the fallback value in the non-responsive input will be used.
But what happens when the fallback/default input is not defined?

Consider this logic here in `_fallbackToKlass()`. Is this logic valid?

This methods effectively says that if a responsive breakpoint deactivates (e.g. `[ngClass.md]`) and the default `[ngClass]` has not been defined, then we should use the `class` attribute value (if any) as the fallback value.

The challenge is the any `ngClass` responsive input could be a string, array, or map. Each do different things.

1) Is this a valid assumption to fallback/restore the initial `class` value?

**Pull request merged**